packages/x-date-pickers/src/internals/hooks/useField/useField.types.ts

```diff
@@ -371,3 +385,21 @@ export type SectionOrdering = {
    */
   endIndex: number;
 };
+
+export interface FieldSlotsComponents {
+  /**
+   * Icon to display inside the clear button.
+   * @default ClearIcon
+   */
+  ClearIcon?: React.ElementType;
+  /**
+   * Icon to display inside the clear button.
+   * @default InputAdornment
+   */
+  InputAdornment?: React.ElementType;
```

Comment:
I've sent a detailed message about this change privately, but TL.DR.: it's probably safer to avoid having this slot. 🤔